### PR TITLE
Refactor display code into DisplayManager singleton

### PR DESCRIPTION
**Pull request by Elkasitu**
With this commit, most if not all of the display logic is refactored
into the display_manager file and most specifically, the DisplayManager
singleton.

* Color pseudo-enum defined in display_manager instead of a dictionary
* Render functionality incorporated into the display_manager instead of
being lone functions in a lone file
* Actors can now take char and color as parameters during
initialization, in order to represent them in the game world

This is done so that the code is clearer and cleaner, all of the display
logic is found in this single file, but also to make the main program be
functional.

### CODE REVIEW

File: displaymanager.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    tdl.set_font('lucida10x10_gs_tc.png', greyscale=True, altLayout=True)
+    console = tdl.Console(SCREEN_WIDTH, SCREEN_HEIGHT)
+    root_console = tdl.init(SCREEN_WIDTH, SCREEN_HEIGHT, title=GAME_TITLE,
+                            fullscreen=False)
+
+    cur_map = None
+    fov_recompute = True
+
+    def gen_map(cls, player):
+        game_map = Dungeon(cls.MAP_WIDTH, cls.MAP_HEIGHT)
```
Comment (Elkasitu) [inline displaymanager.py]: @monzo94 this isn't display logic, but it's whatever... should stay here until we can manage to create a config class of some sort which would hold the big constants above, once that's done we can do the map generation in map and pass the map and the player to the display manager through an init() function or somesuch

File: entities.py [hunk 2]
```diff
@@ -54,29 +55,29 @@ def __init__(self, name, pos, behavior):
...
+        # other stats
+        self._level = 1
+        self._exp = 0
+        self._max_exp = floor(1 + 300 * 2 ** (1/7)) / 4
+        self._gold = 100
         # computed stats
         self._max_hp = self._compute_max_hp()
         self._max_mp = self._compute_max_mp()
+        self._cur_hp = self._max_hp
+        self._cur_mp = self._max_mp
```
Comment (Elkasitu) [inline entities.py]: note to self: undocumented change, perhaps push straight this straight to master as a fix and pull --rebase on this feature branch

File: misc.py [hunk 3]
```diff
@@ -57,3 +57,5 @@ def normalized(self):
         """Return a unitary vector with the same direction."""
         norm = self.norm
         return Vector(self.x / norm, self.y / norm)
+
```
Comment (Elkasitu) [inline misc.py]: deleet dis

File: displaymanager.py (hunk 1 above)
Comment (monzo94) [reply displaymanager.py]: yes, this definitely shouldn't be here

File: entities.py [hunk 4]
```diff
@@ -43,7 +44,7 @@ class Actor(Entity):
     An actor is an entity that can perform actions.
     """
 
-    def __init__(self, name, pos, behavior):
+    def __init__(self, name, pos, behavior, char, color):
```
Comment (monzo94) [inline entities.py]: Shouldn't entities have char and color attributes too? The way I understand it, an entity is anything that has a physical representation in the world.

File: entities.py (hunk 2 above)
Comment (monzo94) [reply entities.py]: Yeah I had to fix the order in the testing branch too cus it was broken.

File: entities.py (hunk 4 above)
Comment (Elkasitu) [reply entities.py]: You are technically correct, the best kind of correct!

**Pull request merged**